test/ref_ops_test.cpp

```diff
@@ -1946,6 +1947,45 @@ TEST_CASE(if_pl_test)
     }
 }
 
+TEST_CASE(isnan_test)
+{
+    // float test
+    {
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        migraphx::shape s{migraphx::shape::float_type, {2, 3}};
+        auto nan_val             = std::numeric_limits<float>::quiet_NaN();
+        std::vector<float> data0 = {1.2, 5.2, nan_val, nan_val, 0., 100.};
+        auto l1                  = mm->add_literal(migraphx::literal{s, data0});
```

Comment:
I think you can do `mm->add_literal(migraphx::literal{s, {1.2, 5.2, nan_val, nan_val, 0., 100.}})`.

Reply:
There looks to be a problem with deducing the vector template type with the nan_val.